code/game/objects/effects/effect_system/effect_system.dm

```diff
@@ -69,6 +69,8 @@ would spawn and follow the beaker, even if it is carried or thrown.
 	var/steps_amt = pick(1,2,3)
 	for(var/j in 1 to steps_amt)
 		sleep(5)
+		if(QDELETED(src) || QDELETED(E))
+			return
```

Comment:
In case of E won't this leave undeleted system around (see autocleanup below)

Reply:
it's to guard against one of many effect types going away?

Reply:
oh I think I misunderstood your comment, are you wanting the total effects counter decremented?

Reply:
or just a qdelete of the entire effects system?

Reply:
Yeah, if only one of the effects get nicked somehow the whole system could possibly stay around when it shouldn't.